plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -315,12 +318,17 @@ private Schema toIcebergSchema(List<Column> columns)
         return new Schema(icebergSchema.asStructType().fields());
     }
 
-    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId)
+    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId, HiveStorageFormat storageFormat)
     {
         if (type instanceof ArrayType || type instanceof MapType || type instanceof RowType) {
             // TODO https://github.com/trinodb/trino/issues/17583 Add support for these complex types
             throw new TrinoException(NOT_SUPPORTED, "Migrating %s type is not supported".formatted(type));
         }
+
+        if (type instanceof TimestampType && storageFormat == AVRO) {
+            throw new TrinoException(NOT_SUPPORTED, "Migrating timestamp type with %s format is not supported.".formatted(storageFormat));
+        }
```

Comment:
Could you file a new issue and leave TODO comment? It would be nice to mention the reason in the issue description.

Reply:
done

Reply:
https://github.com/trinodb/trino/issues/20481